DistroLauncher/algorithms.h

```diff
@@ -18,40 +18,80 @@
 
 /// Common algorithms to be used everywhere in the launcher project with style resembling the std ones.
 
-/// Returns true if [tested] starts with [end]. Null-termination is not required.
-template <typename CharT>
-// NOLINTNEXTLINE(bugprone-easily-swappable-parameters) - Parameters must have the same type.
-bool starts_with(const std::basic_string_view<CharT> tested, const std::basic_string_view<CharT> start)
+namespace algo_internal
 {
-    if (tested.size() < start.size()) {
-        return false;
+    // Tells chars appart from other types
+    template <typename T> constexpr bool is_character() noexcept
+    {
+        return std::is_same_v<T, char> || std::is_same_v<T, wchar_t> || std::is_same_v<T, signed char> ||
+               std::is_same_v<T, unsigned char> || std::is_same_v<T, char16_t> || std::is_same_v<T, char32_t>;
+        // C++20 introduces char8_t
```

Comment:
Did you consider checking if there is a specialization of `std::char_traits<T>::char_type`  to avoid such chain of `is_same_v` and get it automatically up-to-date with the standard library version in use?

Reply:
We could use something like:

```cpp
template <typename CharT>
constexpr bool is_character() {
    // We know for sure double won't have a char_traits specialization.
    using default_type = std::char_traits<double>::int_type;
    return !std::is_same_v<std::char_traits<CharT>::int_type, default_type>;
}
```

[See further details on godbolt.org](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjE/lykAA6oCoRODB7evv5JKWkCIWGRLDFxZgl2mA7pQgRMxASZPn4BldUCtfUEhRHRsfG2dQ1N2a1D3aG9Jf3lAJS2qF7EyOwc5gDMocjeWADUJutuCgTEocAH2CYaAIIbWzuY%2B4dOx8SYrBdXt2abDNteewObicLEMeESnxuXwImBYiQMMKebgIAE9EoxWI83Ah6gAVSHXNAMY6YVSJYi7KKoTy7PAKAD6yBxxCYDliEFm%2BwA7FYbrt%2BbsAPSC3YAdUeAGsGKgAO67fgUhRLR7oRZRei7GUCMAcAi7HEAN0eTF2TPq9JOTEICl2CnRyDwYjwAC8mI4BAA6L4C3ZeVJGXZYKhMLy0AgWtGPA4AEVtBHQIBQzItLOtQNVXnVmAuidC4dR6IOvOuPuOCZQiz1QKB%2BzMZhMAFZkQBJQS7AuPBVI8xmbuHduRvDoCBlxNm4gpq0EBRA7F4nNJ80d2Ye5hsDlF70C0cVrxVw41nsXWk2jvdut9twD9FDkfxsfJy1pw5z4j49bYXOCCPoldrzAcpetb1h%2BjZuAwR7rMWPpvAQSwMLsYBgDudL0gomL0gaQI7uOk7Pm4r7vp%2BIB5j%2BmCkIGmDBqG%2BaRp8UFfCYXLRoxNx5rsoKhIBTHQdu967vubiHnWYEQU2YnIriAAS2AAErYGKCkAOIAPI9kBO5Up4YiJDim58nx5ZoHuQFHuJr4gMBQGoeOrIwsQQLjhc3EHv2ZngdGariFZ1b9jZzJ2bE6ZedmH4bgxBn8jhlamSJ4mtgQlnqb5V7%2BfUgUOc8gjOZyKXAaJchBAIoA%2Ba5qUMrZbKZW4XjEngwBhOguy0MVOX6SWhljjFeXuQwopDlizJJReeVpSyVVAjKuEEDlsX1uJABiLVusNvajRVAUTYcVDLTNYWzO1pb8cZgnCfN4GvrsXAAGyrdZG3pVtbjjjdFqzT1cUXcyuzrGYd3rYym32Y5zK/W9%2B2HZ1AlzaJKGvO8LD/WVJ6A49wOHHDJwI%2B9yO9QtzZBNgABUSNCX5D3jejbj44TRNtRFtzMRw8y0JwDa8H4HBaKQqCcEJljWLaizLFGPw8KQiVc8z8wSiA11mB6N0AJzXesXLXVwv1Kw2kgNvonCSBzmi8LzHC8AoIAaBLxvzHAsAwIgFZwnQsTkJQaDO/QcQGsgiSJJhXBK/SCKYMc9KqNd0g0GGsQWxAUTG6QUShPUKKcOLHtsIIKkMLQadS6QWCgkY4gF/gbzVEaFsF6SVR7qs3N5pgrPcLwtB4FELLECiHhYInJx4IjrfzDtTDAAoABqeCYDKKnopz4v8IIIhiOwUgyIIigqOoBe6AkBhGCg1jWPoHcW5A8yoIk7rEpwgoqU1gqjqYAuWFwXLm83VQ3y4DDuJ4zQ9DBEmMUUoehkipBviMPwCQIH5AYD0UB/QKhf3aAwTowwAHZBQfYG%2BGCJhFD6HECo4xoF6GOF0RBRCJDzAUMLFYegsarB4CzNmRsC6mwjpIDiCgfa7ANIHD0Idji7AgPzKwlgKK4EICQWs6wEi7A8J7WIciuCzF4JLLQswZYgAbFbFuhtSCI0kOsD0AAODQSsuBcDMddBsZiuQNi5JIa6pBObc1NubS21spa2wdhAJAlZEh7jdhAD2iQXbEHCJiTgXCeF8IEUrIRbpQ4EF4JgfARBTgJgSEvYQohxDrzyVvNQic96kBlCyRI6dmb6w4OzNxidTYqT3MEvUqAqC7DiSwXhyB%2BGCOEXqMRqBlEUg2GojRNt5gIHeFgOIHI6mGMRg2a6islYaDMWYzWkguBmEkGYHWjSOGcC8VbTR0tSCyzVuY9YGh1jy3WLYrkGhdZ6xbusdhHiTk%2BK0awjgZhPkm2%2Bec7RpAjTEFSM4SQQA%3D%3D)

Reply:
I played around for a while. Signed and unsined characters fail the test:

https://godbolt.org/z/M17csTEc1

I also tried using an empty struct instead of double but it also has `long` as its integer type. I understand that is the default value.

The concatenated `is_same_v`s is not pretty but it has going for it that it's very transparent as to how it works.

Reply:
I like it better than the previous alternative!